master/cache.lua

```diff
@@ -0,0 +1,465 @@
+local cjson_safe = require "cjson.safe"
+local shmlock = require "resty.lock"
+local http = require "resty.http.simple"
+
+
+local _M = {}
+
+-- In order to make caching code testable, these constants need to be
+-- configurable/exposed through env vars.
+--
+-- Values assigned to these variable need to fufil following condidtion:
+--
+-- CACHE_FIRST_POLL_DELAY << CACHE_EXPIRATION < CACHE_POLL_PERIOD < CACHE_MAX_AGE_SOFT_LIMIT < CACHE_MAX_AGE_HARD_LIMIT
+--
+-- CACHE_BACKEND_REQUEST_TIMEOUT << CACHE_REFRESH_LOCK_TIMEOUT
+--
+-- All are in units of seconds. Below are the defaults:
+local _TIMERS = {}
```

Comment:
True, _TIMERS does not really fit here.

The problem I am facing here is that it's not CONFIG config either, it is a set of default values for various timers. Naming it DEFAULT_VALUES_FOR_TIMERS seems weird.

Would _DEFAULTS work for you? The names of variables here are pretty descriptive already.